arches_lingo/serializers.py

```diff
@@ -49,6 +53,35 @@ class Meta:
         root_node = "appellative_status"
         fields = "__all__"
 
+    def validate(self, data):
+        data = super().validate(data)
+        graph_slug = self.Meta.graph_slug
+        PREF_LABEL_LIST_ITEM = ListItem.objects.get(list_item_values__value="prefLabel")
+
+        if data:
+            new_label_language = data["appellative_status_ascribed_name_language"][0]
+            new_label_type = data["appellative_status_ascribed_relation"][0]
+
+            resource_instance_id = self.instance.resourceinstance.pk
+            resource_instance = self.instance.resourceinstance.__class__.as_model(
+                graph_slug, resource_ids=[resource_instance_id]
+            ).get(pk=resource_instance_id)
+            current_labels = resource_instance.appellative_status
+
+            for label in current_labels:
+                label_language = label.appellative_status_ascribed_name_language[0]
+                label_type = label.appellative_status_ascribed_relation[0]
+                if (
+                    data["tileid"] != label.tileid
+                    and new_label_type["uri"] == PREF_LABEL_LIST_ITEM.uri
+                    and label_type["uri"] == PREF_LABEL_LIST_ITEM.uri
+                    and label_language["uri"] == new_label_language["uri"]
+                ):
+                    raise ValidationError(
+                        "A preferred label with the same language already exists for this scheme."
```

Comment:
I localized this and borrowed the wording from arches-references to prevent creating a second translatable string.